transport/awslambda/wrapper_test.go

```diff
@@ -0,0 +1,473 @@
+		},
+		{
+			errorEncoder: func(
+				ctx context.Context, b string,
+			) {
+			},
+			expectedErrMsg: "errorEncoder takes two arguments, but the second is not error. got string",
+		},
+		{
+			errorEncoder: func(
+				ctx context.Context, err error,
+			) {
+			},
+			expectedErrMsg: "errorEncoder must return two values",
+		},
+		{
+			errorEncoder: func(
+				ctx context.Context, err error,
+			) (apigwResp events.APIGatewayProxyResponse, s string) {
+				apigwResp.Body = `{"error":"yes"}`
+				apigwResp.StatusCode = 500
+				return apigwResp, "nil"
+			},
+			expectedErrMsg: "errorEncoder returns two values, but the second does not implement error",
+		},
+	}
+
+	for _, tc := range testCases {
+		helloHandler := NewServer(
+			makeTest01HelloEndpoint(svc),
+			validDecoder,
+			validEncoder,
+			ServerErrorEncoder(ErrorEncoderWrapper(tc.errorEncoder)),
+		)
+
+		ctx := context.Background()
+		req, _ := json.Marshal(events.APIGatewayProxyRequest{
+			Body: `<err-format>`,
+		})
+
+		_, err := helloHandler.Invoke(ctx, req)
+
+		if err == nil {
+			t.Errorf("\nShould have error")
+		}
+		if err.Error() != tc.expectedErrMsg {
+			t.Fatalf(
+				"\nExpected:\n%+v\nActual:\n%+v",
+				tc.expectedErrMsg, err.Error())
+		}
+	}
+}
+
+func TestWrapperInvalidPayloadFormat(t *testing.T) {
+	svc := serviceTest01{}
+
+	helloHandler := NewServer(
+		makeTest01HelloEndpoint(svc),
+		DecodeRequestWrapper(func(
+			_ context.Context, apigwReq events.APIGatewayProxyRequest,
+		) (helloRequest, error) {
+			request := helloRequest{}
+			err := json.Unmarshal([]byte(apigwReq.Body), &request)
+			return request, err
+		}),
+		EncodeResponseWrapper(func(
+			_ context.Context, response helloResponse,
+		) (apigwResp events.APIGatewayProxyResponse, err error) {
+			respByte, err := json.Marshal(response)
+			if err != nil {
+				return apigwResp, err
+			}
+
+			apigwResp.Body = string(respByte)
+			apigwResp.StatusCode = 200
+			return apigwResp, err
+		}),
+	)
+
+	ctx := context.Background()
+	req := []byte("<invalid-format />")
+	_, err := helloHandler.Invoke(ctx, req)
+
+	if err == nil {
+		t.Fatalf("\nShould have error")
+	}
+}
+
+func TestWrapperErrorInEncodeResponse(t *testing.T) {
+	svc := serviceTest01{}
+
+	helloHandler := NewServer(
+		makeTest01HelloEndpoint(svc),
+		DecodeRequestWrapper(func(
+			_ context.Context, apigwReq events.APIGatewayProxyRequest,
+		) (helloRequest, error) {
+			request := helloRequest{}
+			err := json.Unmarshal([]byte(apigwReq.Body), &request)
+			return request, err
+		}),
+		EncodeResponseWrapper(func(
+			_ context.Context, response helloResponse,
+		) (apigwResp events.APIGatewayProxyResponse, err error) {
+			return apigwResp, fmt.Errorf("error")
+		}),
+	)
+
+	ctx := context.Background()
+	req, _ := json.Marshal(events.APIGatewayProxyRequest{
+		Body: `{"name":"john doe"}`,
+	})
+	_, err := helloHandler.Invoke(ctx, req)
+
+	if err == nil {
+		t.Fatalf("\nShould have error")
```

Comment:
Please don't include `\n` prefixes in t.Log or t.Fatal messages.

Reply:
updated 62efc642e6fd2362e25ce314ca8ad47120405da8